vdr/doc/creator.go

```diff
@@ -47,28 +43,22 @@ type Creator struct {
 
 // didKIDNamingFunc is a function used to name a key used in newly generated DID Documents
 func didKIDNamingFunc(pKey crypto.PublicKey) (string, error) {
-	ecPKey, ok := pKey.(*ecdsa.PublicKey)
-	if !ok {
-		return "", errors.New("could not generate kid: invalid key type")
-	}
-
-	if ecPKey.Curve == nil {
-		return "", errors.New("could not generate kid: empty key curve")
-	}
-
 	// according to RFC006:
 	// --------------------
 
 	// generate idString
-	pkBytes := elliptic.Marshal(ecPKey.Curve, ecPKey.X, ecPKey.Y)
-	pkHash := sha256.Sum256(pkBytes)
-	idString := base58.Encode(pkHash[:], base58.BitcoinAlphabet)
-
-	// generate kid fragment
 	jwKey, err := jwk.New(pKey)
+	if err != nil {
+		return "", fmt.Errorf("could not generate kid: invalid key type")
+	}
+
+	pkHash, err := jwKey.Thumbprint(crypto.SHA256)
 	if err != nil {
 		return "", err
 	}
+	idString := base58.Encode(pkHash[:], base58.BitcoinAlphabet)
```

Comment:
make a const of `BitcoinAlphabet` since it's now used twice as JWK thumbprint encoding scheme (or make a util func)

Reply:
moved Thumbprint to crypto and added test that matches example from RFC